src/moonlink/Cargo.toml

```diff
@@ -8,6 +8,7 @@ license = { workspace = true }
 default = ["storage-fs"]
 storage-all = ["storage-fs", "storage-s3", "storage-gcs"]
 test-utils = []
+rest-catalog = ["iceberg-catalog-rest"]
```

Comment:
We should have one unified rest catalog feature: https://github.com/Mooncake-Labs/moonlink/blob/94656f903a8fb91565b6fe41b7ebcdef6d9729e8/src/moonlink/Cargo.toml#L34